src/App.js

```diff
@@ -1,25 +1,107 @@
-import logo from './logo.svg';
 import './App.css';
+import {useMemo,useState } from "react";
 
-function App() {
+function Square(props) {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+      <button className="square" onClick={props.onClick}>
+        {props.value}
+      </button>
+    );
 }
 
-export default App;
+function Board(props) {
+  
+    // const rendersquare = props.squares.map((item ,index) =>
+    //   <Square key={index} 
+    //     value={item} onClick={() => props.handleClick(index)} />
+    // );
+    // console.log("test",props.squares);
+    // const rendersquare = (param1, param2) => {
+    //   const num1 = props.squares.slice(param1, param2);
+    //   return <>
+    //     { num1.map((item ,index) => <Square key={index} data={item}  /> )} < />
+    // };
+    
+    // const boardrow = (num1, num2) => {
+    //   return (
+    //     <div className="board-row">
+    //         {rendersquare.slice(num1, num2)}
+    //     </div>
+    // )};
+  
+    return (
+      <div>
+        <h1>Tic Tac Toe</h1>
+        <div className="status">{props.turn}</div>
+        <div className="board">
+          {props.squares.map((item, index) => {
+            return <Square key={index} value={item} onClick={() => props.handleClick(index)} />
+          })}
+        </div>
+          {/* {boardrow(0, 3)}
+          {boardrow(3, 6)}
+          {boardrow(6, 9)} */}
+        <button className="restart" onClick={props.handleRestart}>Restart Game</button>
+        </div>
+    );
+  }
+  
+  function Game(props) {
+    const [xIsNext, setxIsNext] = useState(true);
+    const [box, setBox] = useState(Array(9).fill(""));
+    const winner = useMemo(() => calculateWinner(box), [xIsNext]);
+    let turn ;
+    
+    const handleClick = (i) => {
+      if (calculateWinner(box) || box[i]) {
+        return;
+      }
+      box[i] = xIsNext ? "X" : "O" ;
+      setBox(box) ;
+      setxIsNext(!xIsNext) ;
+    }
+
+    if (winner) {
+      turn = winner + " is Winner!" ;
+    }
+    else if (!winner && box.includes("") === false) {
+      turn = "Drawn!!!" ;
+    }
+    else {
+      turn = (xIsNext ? "X" : "O") + " Play" ;
+    }
+    
+    const handleRestart = () => {
+      setxIsNext(true);
+      setBox(Array(9).fill(""));
+    }
+    
+  
+      return (
+        <Board  squares={box} turn={turn} handleClick={handleClick} handleRestart={handleRestart}  />
```

Comment:
Usually, when you define a function callback you would name them `handleXXXX` but when you want to access them as props you will name `onXXXXX`

eg: 
```jsx
<Board squares={box} turn={turn} onClick={handleClick} onRestart={handleRestart}  />
```

ps: personally I just use `onXXXXX` on both times when I define and access them as props as it is easier to find and I'm too lazy to change the name 😛